src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Documents/VisualStudioFileChangeTrackerFactory.cs

```diff
@@ -25,10 +25,17 @@ internal class VisualStudioFileChangeTrackerFactory : IFileChangeTrackerFactory
         ProjectSnapshotManagerDispatcher dispatcher,
         IErrorReporter errorReporter)
     {
-        _errorReporter = errorReporter;
-        _fileChangeService = (IVsAsyncFileChangeEx)serviceProvider.GetService(typeof(SVsFileChangeEx));
-        _projectSnapshotManagerDispatcher = dispatcher;
+        _dispatcher = dispatcher;
         _joinableTaskContext = joinableTaskContext;
+        _errorReporter = errorReporter;
+
+        var jtf = _joinableTaskContext.Factory;
+        _getFileChangeServiceTask = jtf.RunAsync(async () =>
+        {
+            await jtf.SwitchToMainThreadAsync();
+
+            return (IVsAsyncFileChangeEx)serviceProvider.GetService(typeof(SVsFileChangeEx));
+        });
```

Comment:
This can be simplified if you use `[Import(typeof(SAsyncServiceProvider))] IAsyncServiceProvider serviceProvider,` , and `jtf.RunAsync(() => serviceProvider.GetServiceAsync...`

I believe that will also make it "smarter" but don't hold me to that :P

Reply:
Good suggestion - thanks!